internal/services/network/custom_ip_prefix_resource_test.go

```diff
@@ -0,0 +1,256 @@
+package network_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CustomIpPrefixResource struct {
+}
```

Comment:
since this is creating an IP Prefix, can we create a resource *from* this Resource? that would allow us to test this works end-to-end

Reply:
@tombuildsstuff  Thanks for you feedback. For the comment above, do you mean to add a test case to create a `azurerm_public_ip_prefix` from this resource `azurerm_custom_ip_prefix`? If so, that is another feature requested from service team. That is "[Create a public IP prefix from a custom IP prefix](https://docs.microsoft.com/en-us/azure/virtual-network/ip-services/manage-custom-ip-address-prefix#create-a-public-ip-prefix-from-a-custom-ip-prefix)". Since this feature requires Terraform to support the [ip_prefix](https://github.com/hashicorp/terraform-provider-azurerm/blob/10a2ed50f79f1789320e76b775f38abd23d2c86b/internal/services/network/public_ip_prefix_resource.go#L85) of `azurerm_public_ip_prefix` setting, I plan to implement support it after this PR is merged. The end-to-end test  about creating a resource from this Resource would be added to `azurerm_public_ip_prefix` test at that time. WDYT?